src/data_provider/src/sysInfoMac.cpp

```diff
@@ -99,48 +102,83 @@ nlohmann::json SysInfo::getHardware() const
 
 static void getPackagesFromPath(const std::string& pkgDirectory, const int pkgType, std::function<void(nlohmann::json&)> callback)
 {
-    const auto packages { Utils::enumerateDir(pkgDirectory) };
-
-    for (const auto& package : packages)
+    if (MACPORTS == pkgType)
     {
-        if (PKG == pkgType)
+        if (Utils::existsRegular(pkgDirectory + "/" + MACPORTS_DB_NAME))
         {
-            if (Utils::endsWith(package, ".app"))
+            try
             {
-                nlohmann::json jsPackage;
-                FactoryPackageFamilyCreator<OSPlatformType::BSDBASED>::create(std::make_pair(PackageContext{pkgDirectory, package, ""}, pkgType))->buildPackageData(jsPackage);
+                std::shared_ptr<SQLite::IConnection> sqliteConnection = std::make_shared<SQLite::Connection>(pkgDirectory + "/" + MACPORTS_DB_NAME);
+
+                SQLite::Statement stmt
+                {
+                    sqliteConnection,
+                    MACPORTS_QUERY
+                };
 
-                if (!jsPackage.at("name").get_ref<const std::string&>().empty())
+                std::pair<SQLite::IStatement&, const int&> pkgContext {std::make_pair(std::ref(stmt), std::cref(pkgType))};
+
+                while (SQLITE_ROW == stmt.step())
                 {
-                    // Only return valid content packages
-                    callback(jsPackage);
+                    nlohmann::json jsPackage;
+                    FactoryPackageFamilyCreator<OSPlatformType::BSDBASED>::create(pkgContext)->buildPackageData(jsPackage);
+
+                    if (!jsPackage.at("name").get_ref<const std::string&>().empty())
+                    {
+                        // Only return valid content packages
+                        callback(jsPackage);
+                    }
                 }
             }
+            catch (...)
+            {}
         }
-        else if (BREW == pkgType)
+    }
+    else
+    {
+        const auto packages { Utils::enumerateDir(pkgDirectory) };
+
+        for (const auto& package : packages)
         {
-            if (!Utils::startsWith(package, "."))
+            if (PKG == pkgType)
```

Comment:
This recurrent and unnecessary conditional inside the loop could be avoided if the whole selection of "pkgType" is done with an external unique switch case.
In my opinion, this change also would make code cleaner.

Reply:
Probably we can improve that in the future, but that is not part of this PR I prefer to not change legacy code more than needed for this feature.